src/components/Form/Form.js

```diff
@@ -11,96 +11,102 @@ const SUBMITTING = 'SUBMITTING';
 const SUCCESS = 'SUCCESS';
 const ERROR = 'ERROR';
 
-export const Form = ({
-  allowSubmit,
-  error,
-  fields,
-  form,
-  onChange,
-  onSubmit,
-  submitting,
-  success,
-  ...props
-}) => {
-  const [status, setStatus] = React.useState(NONE);
-  const [fieldState, setFieldState] = React.useState(
-    fields.reduce((acc, { defaultValue, name }) => {
-      acc[name] = typeof defaultValue !== 'undefined' ? defaultValue : '';
-      return acc;
-    }, {}),
-  );
-
-  const fieldData = fields.map((field) => ({
-    ...field,
-    onChange: (e) => {
-      // have to pull the value from the synthetic event here, because react throws it away
-      const value = e.target.value;
-      if (value !== fieldState[field.name]) {
-        setFieldState((prev) => ({
-          ...prev,
-          [field.name]: value,
-        }));
-      }
+export const Form = React.forwardRef(
```

Comment:
FYI using an arrow function in `forwardRef` loses the component name, because it can no longer infer name from the assigned variable. I prefer to use `React.forwardRef(function Thing(){...})` for that reason. That's a nit tho